gateway-server/src/main/java/org/apache/knox/gateway/services/token/impl/DefaultTokenStateService.java

```diff
@@ -43,9 +48,19 @@
 
   private final Map<String, Long> maxTokenLifetimes = new HashMap<>();
 
+  /* token eviction interval in seconds, default is 5 mins */
+  private long tokenEvictionInterval;
+
+  private final ScheduledExecutorService evictionScheduler = Executors.newScheduledThreadPool(1);
+
 
   @Override
   public void init(final GatewayConfig config, final Map<String, String> options) throws ServiceLifecycleException {
+    tokenEvictionInterval = config.getKnoxTokenEvictionInterval();
+    if (tokenEvictionInterval > 0) {
```

Comment:
This probably should go in start() method.

Reply:
Yea agreed should go in start.